src/catalog-debug/src/main.rs

```diff
@@ -689,10 +705,12 @@ async fn upgrade_check(
             .latest_schema::<SourceData, (), Timestamp, Diff>(shard_id, diagnostics)
             .await
             .expect("invalid persist usage");
-        // We should always have schemas registered for Shards, unless their environment happened
-        // to crash after running DDL and hasn't come back up yet.
+        // If in the new version a BuiltinTable or BuiltinSource is changed (e.g. a new
+        // column is added) then we'll potentially have a new shard, but no writes will
+        // have occurred so no schema will be registered.
         let Some((_schema_id, persisted_relation_desc, _)) = persisted_schema else {
-            anyhow::bail!("no schema found for {gid}: {shard_id}, did their environment crash?");
+            println!("no schema found for {gid} '{shard_id}', continuing...");
+            continue;
```

Comment:
Are there any repercussions of not `anyhow::bail`ing if if a user's persist relation doesn't have a shard?

Reply:
There shouldn't be! The only other case that this might occur is if a shard was created (e.g. a table was created) and then `environmentd` immediately crashed. Although on restart `environmentd` will register a schema, so we should be fine here.